tachyon/node/base/node_internals.h

```diff
@@ -484,6 +484,31 @@ class CppValueTraits<base::CppStackValue<T>> {
   }
 };
 
+template <typename T>
+class CppValueTraits<std::optional<T>> {
+ public:
+  static bool ToNativeValue(const Napi::Value& value, std::optional<T> v) {
+    if (value.IsUndefined() || value.IsNull())
+      *v = std::nullopt;
+    else
+      *v = reinterpret_cast<T>(value);
+    return true;
+  }
+
+  static Napi::Value ToJSValue(const Napi::CallbackInfo& info,
+                               const std::optional<T> value) {
+    if (!value)
+      return Napi::Value(info.Env(), NULL);
```

Comment:
Please avoid using `NULL` in c++.

```suggestion
      return Napi::Value(info.Env(), nullptr);
```